### DISCUSSION TITLE
Trying to make a selection rect with mouse in customtkinter (+ some other questions)

### DISCUSSION THREAD

**Discussion by Sobrtonog** [TomSchimansky/CustomTkinter#1988] [Q&A]
![ctkDragRect](https://github.com/TomSchimansky/CustomTkinter/assets/7352665/dfefaee0-9983-4dc9-b33e-b8bfc3ea2f05)

Hey there folks, been trying to make a selection rect in ctk - this is what I have so far and it seems... decent although I feel like I'm hitting some limitations when it comes to a few more baseline features I want it to have. It's basically some mouse bind events + I'm resizing a label frame along with re-anchoring the label depending on drag coordinate delta.

- Can transparent widgets actually be transparent ? transparency color seems to depend on the baseline widget it's attached to (if you see my gif above, I'm using a label as the selection rect with transparent color but it still renders color as if its opaque) I want my selection rect to be transparent over the image texture I have below it.
- Can we give labels a border + borderColor kind of like frames? I was using frames previously, but it seemed pretty jittery in motion. Also CTkImage is verrrryyy buggy when trying to use it as a selectionRect + it doesn't do transparency so that's a no go too.

Or if this is not the right way to go about it, does anyone have any better ideas?

Thanks in advance.

**Comment by Akascape** [comment]
@Sobrtonog For this type of things, you have to draw all the components in the canvas of tkinter. The widgets that you place cannot be transparent, but instead you can draw a rectangular box with dotted lines and also add bindings to it. 

Like for example, I made a node based ui system in tkinter using a single canvas, the background has an image of a grid and the other components are drawn in top of it.

 https://github.com/Akascape/TkNodeSystem

**Reply by Sobrtonog** [reply]
Hey @Akascape thanks for the answer, I already figured as much and as cumbersome qt is I've kind of resigned myself to it for now.

I love ctk and would much prefer it to use it over qt, but I need to be able to layer many transparent obj's on top of each other for my next project.
